tests/foreman/cli/test_user.py

```diff
@@ -295,20 +214,22 @@ def test_negative_create_with_invalid_username(self):
 
         @Assert: User is not created. Appropriate error shown.
         """
-        for invalid_name in ('',
-                             'space {0}'.format(gen_string('alpha')),
-                             gen_string('alpha', 101),
-                             gen_string('html')):
-            with self.subTest(invalid_name):
+        invalid_names = (
+            '', 'space {0}'.format(gen_string('alpha')),
+            gen_string('alpha', 101),
+            gen_string('html')
+        )
+        for invalid_name in invalid_names:
+            with self.subTest(invalid_name), self.assertRaises(
+                    CLIReturnCodeError):
```

Comment:
If just the create call was within the nested context manager then it would be fine.